### ISSUE TITLE
Crash when lost internet connection

### ISSUE THREAD

**Issue by ppamorim** [jgh-/VideoCore#291]
I verified the sample of your source code and noticed you don't handle the lost of internet connection when the application is streaming. It causes a bug and the app crashes.

I used the `VCSessionDelegate` to get a callback status of the stream, when the app lost the connection, the library calls the delegate and returns the status. After all, crashes.

**Comment by jamolama** [comment]
@ppamorim are you calling endRtmpSession for your VCSimpleSession after receiving the error? Calling endRtmpSession when the most recent status is VCSessionStatusError or VCSessionStatusEnded will cause a crash.

**Comment by ppamorim** [comment]
Yes, but if I let these callbacks empty, another error happens. I'll test this problem on the sample, say connected.

**Comment by ppamorim** [comment]
Well done, I cannot edit/compile the sample project because it has errors.

**Comment by ppamorim** [comment]
When I do this:

``` swift

  func connectionStatusChanged(sessionState: VCSessionState) {
    if sessionStream != nil {
      switch sessionStream!.rtmpSessionState {

      case .Starting:
        //Yeheee loading camera!
        break

      case .PreviewStarted:
        //Yeeeep started!
        break

      case .Ended:
        //Nothing
        break

      case .Error:
        self.sessionStream?.endRtmpSession()
        break
      }
    }
  }
```

Sometimes the app crashes, sometimes not, randomly.

But when the app crashes, with or without the code `self.sessionStream?.endRtmpSession()`, happens:

![With code](https://cloud.githubusercontent.com/assets/5521682/14029395/9d79a386-f1e0-11e5-824d-11d5484576ab.png)

This error is a huge problem, the app cannot crash with this simple problem.

**Comment by ppamorim** [comment]
I discovered a solution:

```
Simple, remove the log.
```
